platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/JsonBundle.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.migratable;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.Map;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.Version;
+
+class JsonBundle {
+  private final String name;
+
+  private final Version version;
+
+  private final long id;
+
+  private final int state;
+
+  private final String location;
+
+  JsonBundle(Map<String, Object> binfo) {
+    this.name = JsonUtils.getStringFrom(binfo, ProfileMigratable.PROFILE_NAME, true);
+    this.id = JsonUtils.getLongFrom(binfo, ProfileMigratable.PROFILE_ID, true);
+    this.state = JsonUtils.getIntegerFrom(binfo, ProfileMigratable.PROFILE_STATE, true);
+    this.location = JsonUtils.getStringFrom(binfo, ProfileMigratable.PROFILE_LOCATION, true);
+    final List<Object> vinfo =
+        JsonUtils.getListFrom(binfo, ProfileMigratable.PROFILE_VERSION, true);
+
+    this.version =
+        new Version( //
+            JsonUtils.getIntegerFrom(vinfo, 0, true),
+            JsonUtils.getIntegerFrom(vinfo, 1, true),
+            JsonUtils.getIntegerFrom(vinfo, 2, true),
+            JsonUtils.getStringFrom(vinfo, 3, true));
+  }
+
+  @VisibleForTesting
+  JsonBundle(String name, Version version, long id, int state, String location) {
+    this.name = name;
+    this.version = version;
+    this.id = id;
+    this.state = state;
+    this.location = location;
+  }
+
+  public static String getName(Bundle bundle) {
+    return bundle.getSymbolicName() + '/' + bundle.getVersion();
+  }
+
+  public static SimpleState getSimpleState(Bundle bundle) {
+    return JsonBundle.getSimpleState(bundle.getState());
+  }
+
+  private static SimpleState getSimpleState(int state) {
+    if (state == Bundle.UNINSTALLED) {
+      return SimpleState.UNINSTALLED;
+    } else if ((state == Bundle.STARTING) || (state == Bundle.ACTIVE)) {
```

Comment:
If a bundle is Starting, but fails to become Active (fails for some reason), we would still return Active here.  Is that what we want? https://osgi.org/javadoc/r4v41/org/osgi/framework/Bundle.html#STARTING

Reply:
This would be on the original node. First it would mean you are running import on a system that is not ready yet which is not advisable to start with. Second, we would still try to start it on on the destination side. On the destination side, STARTING or STOPPING can only be reported when the start() or the stop() method of the bundle is being called. When it returns, per documentation, it would report one of the other "stable" state which means we should not see those when we come back on a subsequent pass to verify the state of things.